tests/tape/test_qscript.py

```diff
@@ -186,17 +186,153 @@ def test_update_par_info_update_trainable_params(self):
+        assert p_i[7] == {"op": m[0].obs, "op_idx": 0, "p_idx": 0}
+
+        assert qs._trainable_params == list(range(8))
+
+    def test_get_operation(self):
+        """Tests the tape method get_operation"""
+        ops = [
+            qml.RX(1.2, wires=0),
+            qml.Rot(2.3, 3.4, 5.6, wires=0),
+            qml.PauliX(wires=0),
+            qml.QubitUnitary(np.eye(2), wires=0),
+            qml.U2(-1, -2, wires=0),
+        ]
+        m = [qml.expval(qml.Hermitian(2 * np.eye(2), wires=0))]
+        qs = QuantumScript(ops, m)
+
+        with pytest.warns(
+            UserWarning,
+            match="The get_operation will soon be updated to also return the index of the trainable operation in the tape.",
+        ):
+            op_0, p_id_0 = qs.get_operation(0)
+            assert op_0 == ops[0] and p_id_0 == 0
+
+            op_1, p_id_1 = qs.get_operation(1)
+            assert op_1 == ops[1] and p_id_1 == 0
+
+            op_2, p_id_2 = qs.get_operation(2)
+            assert op_2 == ops[1] and p_id_2 == 1
+
+            op_3, p_id_3 = qs.get_operation(3)
+            assert op_3 == ops[1] and p_id_3 == 2
+
+            op_4, p_id_4 = qs.get_operation(4)
+            assert op_4 == ops[3] and p_id_4 == 0
+
+            op_5, p_id_5 = qs.get_operation(5)
+            assert op_5 == ops[4] and p_id_5 == 0
+
+            op_6, p_id_6 = qs.get_operation(6)
+            assert op_6 == ops[4] and p_id_6 == 1
+
+            _, p_id_0 = qs.get_operation(7)
+        assert p_id_0 == 0
+
+    def test_get_operation_return_index(self):
+        """Tests the tape method get_operation with `return_op_index` bool."""
+        ops = [
+            qml.RX(1.2, wires=0),
+            qml.Rot(2.3, 3.4, 5.6, wires=0),
+            qml.PauliX(wires=0),
+            qml.QubitUnitary(np.eye(2), wires=0),
+            qml.U2(-1, -2, wires=0),
+        ]
+        m = [qml.expval(qml.Hermitian(2 * np.eye(2), wires=0))]
+        qs = QuantumScript(ops, m)
+
+        op_0, op_id_0, p_id_0 = qs.get_operation(0, True)
+        assert op_0 == ops[0] and op_id_0 == 0 and p_id_0 == 0
+
+        op_1, op_id_1, p_id_1 = qs.get_operation(1, True)
+        assert op_1 == ops[1] and op_id_1 == 1 and p_id_1 == 0
+
+        op_2, op_id_2, p_id_2 = qs.get_operation(2, True)
+        assert op_2 == ops[1] and op_id_2 == 1 and p_id_2 == 1
+
+        op_3, op_id_3, p_id_3 = qs.get_operation(3, True)
+        assert op_3 == ops[1] and op_id_3 == 1 and p_id_3 == 2
+
+        op_4, op_id_4, p_id_4 = qs.get_operation(4, True)
+        assert op_4 == ops[3] and op_id_4 == 3 and p_id_4 == 0
+
+        op_5, op_id_5, p_id_5 = qs.get_operation(5, True)
+        assert op_5 == ops[4] and op_id_5 == 4 and p_id_5 == 0
+
+        op_6, op_id_6, p_id_6 = qs.get_operation(6, True)
+        assert op_6 == ops[4] and op_id_6 == 4 and p_id_6 == 1
+
+        _, obs_id_0, p_id_0 = qs.get_operation(7, True)
+        assert obs_id_0 == 0 and p_id_0 == 0
+
+    def test_get_operation_(self):
```

Comment:
```suggestion
    def test_get_operation(self):
```

But the same name once again. For the name, include details about how this test *differs* from the other ones. What exact property of `get_operation` is it testing?

Reply:
Here it is testing the private method and this test will replace the `test_get_operation` in the future.

Reply:
How about `test_private_get_operation`?

Reply:
Yes I already change it :+1: